community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveSortedArraySet.java

```diff
@@ -21,26 +21,43 @@
 
 import java.util.Arrays;
 
+/**
+ * Specialized methods for operations on sets represented as sorted primitive arrays.
+ *
+ * This class does not contain a complete set of operations for all primitives, but only
+ * the ones that were needed. Feel free to add specializations on demand, but remember to test.
+ */
 public class PrimitiveSortedArraySet
```

Comment:
would be nice to know if results returned from methods in this class also returns arrays that are actually sets